src/main.js

```diff
@@ -233,6 +233,13 @@ define('peaks', [
      * @type {Function}
      */
     this.logger = console.error.bind(console);
+
+    /**
+     * Destroy anything that needs to be destoryed
+     */
+    this.destroy = function() {
+      this.waveform.destroy();
```

Comment:
wouldn't you want to make sure this.waveform is defined prior to call destroy() on it?